src/ResourceManager/Cdn/Commands.Cdn/Profile/GetAzureRmCdnProfileSupportedOptimizationType.cs

```diff
@@ -0,0 +1,59 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Cdn.Common;
+using Microsoft.Azure.Commands.Cdn.Models.Profile;
+using Microsoft.Azure.Commands.Cdn.Properties;
+using Microsoft.Azure.Management.Cdn;
+using System.Linq;
+using Microsoft.Azure.Commands.Cdn.Helpers;
+using Microsoft.Azure.Commands.Cdn.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Cdn.Profile
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmCdnProfileSupportedOptimizationType", DefaultParameterSetName = FieldsParameterSet), OutputType(typeof(List<PSOptimizationType>))]
```

Comment:
@prakharsharma10 rather than having this separate cmdlet to get additional information about a profile, we should see if we can add a `SwitchParameter` to the existing `Get-AzureRmCdnProfile` cmdlet that the user will pass to signal they want the cmdlet to make the extra API call to get information about the supported optimization types, and that can be found in an additional property of the `PSProfile` object that is returned by that cmdlet. Let me know what you think

Reply:
Thanks for the suggestion :) . It's a design decision and we have gone through this in the past here https://github.com/Azure/azure-rest-api-specs/pull/1523/files/ef75b2c5a9a237c413122f141c4b6c3722ac103d#diff-6bb303ad17bff3c22d65d5ab3ee7b8cf . Since the supported optimization types can change without any update from the user side, we want to keep it separate from the profile entity. For example, if we return the profile along with supported optimization types, the profile as a resource will keep changing from time to time as we change the types for the SKUs.